Signal/src/ViewControllers/ConversationView/Cells/OWSMessageCell.m

```diff
@@ -269,11 +269,10 @@ - (BOOL)updateAvatarView
     }
 
     TSIncomingMessage *incomingMessage = (TSIncomingMessage *)self.viewItem.interaction;
-    OWSAvatarBuilder *avatarBuilder = [[OWSContactAvatarBuilder alloc] initWithSignalId:incomingMessage.authorId
-                                                                                  color:self.conversationStyle.primaryColor
-                                                                               diameter:self.avatarSize
-                                                                        contactsManager:contactsManager];
-    self.avatarView.image = [avatarBuilder build];
+    TSThread *authorThread = [TSContactThread getOrCreateThreadWithContactId:incomingMessage.authorId];
```

Comment:
We could look up the thread for this author in the `ConversationViewItem`, so that we a) use the right db connection b) avoid sneaky read transaction here.

Reply:
This is a sneaky read/__write__ transaction - seems like something we should avoid in UI code.

Reply:
Done.